kernel/src/path.rs

```diff
@@ -299,6 +302,50 @@ impl ParsedLogPath<Url> {
         }
         Ok(path)
     }
+
+    /// Extract the In-Commit Timestamp from the CommitInfo action in this commit log file.
+    /// This is a utility function that can be used by multiple parts of the codebase
+    /// (snapshot, CDF, time travel, etc.).
+    ///
+    /// Returns the inCommitTimestamp value, or an error if ICT is not found or cannot be read.
+    /// Callers should handle enablement version checks before calling this method.
+    #[internal_api]
+    pub(crate) fn read_in_commit_timestamp(&self, engine: &dyn Engine) -> DeltaResult<i64> {
```

Comment:
i'm a little torn on whether or not this really makes sense as a method here or just a static fn

Reply:
This is fine as a method since we always need the ParsedLogPath to compute it. In fact, I'd originally had this as a static method and you suggested I make this a method under ParsedLogPath lol